src/middlewared/middlewared/plugins/truenas_connect/update.py

```diff
@@ -228,8 +228,14 @@ async def unset_registration_details(self):
             await self.middleware.call('certificate.setup_self_signed_cert_for_ui')
             logger.debug('Restarting nginx to consume self generated cert')
             await self.middleware.call('system.general.ui_restart', 2)
-            logger.debug('Revoking existing TNC cert')
-            await self.middleware.call('tn_connect.acme.revoke_cert')
+            if skip_revoking_cert_and_account is False:
```

Comment:
This is really hard logic to read because it's a double-negative. Should name the keyword argument something else. (i.e.
```
if revoke_cert_and_account:
    logger.debug('Revoking existing...
    ....


if not revoke_cert_and_account:
    ...
```